File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int ptr1 = 0;
        int ptr2 = 0;
        int n = s.size();

        for(int i=n-1; i>=0; i--){
            if(isalpha(s[i])){
                ptr1 = i+1;
                break;
            }
        }
        
        for(int i=ptr1-1; i>=0; i--){
            if(s[i] == ' '){
                ptr2 = i;
                break;
            }
            if(i==0){
                ptr2 = i-1;
                break;
            }
        }
        int result = ptr1-ptr2-1;
        return result;
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        ptr1 = 0
        ptr2 = 0
        n = len(s)

        for i in range(n-1,-1,-1):
            if s[i].isalpha():
                ptr1 = i+1
                break
        
        for i in range(ptr1-1,-1,-1):
            if s[i] == " ":
                ptr2 = i
                break
            if i==0:
                ptr2 = i-1
                break
        
        result = ptr1-ptr2-1
        return result